specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-03-01/examples/dataConnectors/CreateThreatIntelligenceTaxiiDataConnector.json

```diff
@@ -0,0 +1,81 @@
+{
+  "parameters": {
+    "api-version": "2024-03-01",
+    "subscriptionId": "d0cfe6b2-9ac0-4464-9919-dccaee2e48c0",
+    "resourceGroupName": "myRg",
+    "workspaceName": "myWorkspace",
+    "dataConnectorId": "73e01a99-5cd7-4139-a149-9f2736ff2ab5",
+    "dataConnector": {
+      "etag": "d12423f6-a60b-4ca5-88c0-feb1a182d0f0",
+      "kind": "ThreatIntelligenceTaxii",
+      "properties": {
+        "tenantId": "06b3ccb8-1384-4bcc-aec7-852f6d57161b",
+        "taxiiServer": "https://limo.anomali.com/api/v1/taxii2/feeds",
+        "collectionId": "135",
+        "workspaceId": "dd124572-4962-4495-9bd2-9dade12314b4",
+        "friendlyName": "testTaxii",
+        "userName": "--",
+        "password": "--",
+        "taxiiLookbackPeriod": "2023-01-01T13:00:30.123Z",
+        "pollingFrequency": "OnceADay",
+        "dataTypes": {
+          "taxiiClient": {
+            "state": "Enabled"
+          }
+        }
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/d0cfe6b2-9ac0-4464-9919-dccaee2e48c0/resourceGroups/myRg/providers/Microsoft.OperationalInsights/workspaces/myWorkspace/providers/Microsoft.SecurityInsights/dataConnectors/73e01a99-5cd7-4139-a149-9f2736ff2ab5",
+        "name": "73e01a99-5cd7-4139-a149-9f2736ff2ab5",
+        "type": "Microsoft.SecurityInsights/dataConnectors",
+        "etag": "d12423f6-a60b-4ca5-88c0-feb1a182d0f0",
+        "kind": "ThreatIntelligenceTaxii",
+        "properties": {
+          "tenantId": "06b3ccb8-1384-4bcc-aec7-852f6d57161b",
+          "taxiiServer": "https://limo.anomali.com/api/v1/taxii2/feeds",
+          "collectionId": "135",
+          "workspaceId": "28e5f051-34cb-4208-9037-693e5342a871",
+          "friendlyName": "testTaxii",
+          "userName": null,
```

Comment:
You shouldn't actually return null as a value unless its annotated x-ms-nullable:true, which is usually discouraged anhway (you can just not return a field in a payload if its optional)